examples/common/example.py

```diff
@@ -125,11 +135,26 @@ def _get_example_settings(profile: Dict, example_abs_path: str) -> Dict:
     exception. It is up to each individual module to check whether the settings are
     complete.
     """
-    example_settings = profile["examples"]
+    example_settings = None
+    if "examples" in profile:
+        example_settings = profile["examples"]
+
+    # Prefer profile specific settings to global defaults
+    if "examples" in globals:
+        global_settings = globals["examples"]
+        if not example_settings:
```

Comment:
I'd recommend switching round this `if not...else` so it's just an `if...else`, since this is already a complex code block. Same goes for the `if not in...else` further down.

Reply:
Revised